src/main/java/com/flightstats/hub/dao/aws/ClusterContentService.java

```diff
@@ -134,16 +139,43 @@ public boolean historicalInsert(String channelName, Content content) throws Exce
         if (channel.isSingle()) {
             content = s3SingleContentDao.get(channelName, key);
         } else if (channel.isBatch()) {
-            content = s3BatchContentDao.get(channelName, key);
+            content = spokeReadContentDao.get(channelName, key);
+            if (content == null) {
+                content = getFromS3BatchAndWriteToSpokeBatch(channelName, key);
+            }
         } else {
-            content = s3SingleContentDao.get(channelName, key);
+            content = spokeReadContentDao.get(channelName, key);
+            if (content == null) {
```

Comment:
pondering the relative ordering of the s3Single get and the S3Batch get...

Reply:
The channel must be ```BOTH``` since its not ```SINGLE``` or ```BATCH```. So the idea is to check nearer, cheaper stores first. Read cache > S3Single > S3Batch (storing the result in the read cache).